tests/utils/test_annotations.py

```diff
@@ -133,3 +185,121 @@ def test_deprecated_and_keyword_second():
     Returns:
         y"""  # noqa: E501
     )
+
+
+def test_deprecated_class():
+    msg = "``tests.utils.test_annotations.DeprecatedClass`` is deprecated"
+    with pytest.warns(FutureWarning, match=re.escape(msg)):
+        DeprecatedClass()
+    assert msg in DeprecatedClass.__doc__
+
+
+def test_deprecated_class_method():
+    msg = "``tests.utils.test_annotations.DeprecatedClass`` is deprecated"
+    with pytest.warns(FutureWarning, match=re.escape(msg)):
+        instance = DeprecatedClass()
+        assert instance.greet() == "Hello"
```

Comment:
```suggestion
    assert instance.greet() == "Hello"
```

to clarify what throws warning.